clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/actions/SubmitTitusJob.java

```diff
@@ -102,6 +105,23 @@ public Result apply(@Nonnull SubmitTitusJobCommand command, @Nonnull Saga saga)
                         TitusJobNameResolver.resolveJobName(titusClient, description);
                     saga.log("Retrying with job name %s", nextServerGroupName[0]);
                   }
+                  // Can't do an exact match on the server group name because the error from Titus
+                  // adds dashes if the server group name does not contain a stack or free form
+                  // details, i.e. testapp---v001
+                  else if (statusDescription != null
+                      && statusDescription.contains(
+                          "Constraint violation - job with group sequence")
+                      && statusDescription.contains("exists")
+                      && submissionAttempts.intValue() > 1) {
+
+                    String jobId = titusClient.findJobByName(nextServerGroupName[0]).getId();
```

Comment:
Thought about parsing the ID out of the exception message but decided against relying too heavily on the format of the message.  If this call fails it will be retried as part of this retry support lambda.